sdk/aqueduct/artifacts/bool_artifact.py

```diff
@@ -2,17 +2,18 @@
 
 import json
 import uuid
-from typing import Any, Dict, Optional
+from typing import Any, Dict, Optional, Union
 
-from aqueduct.dag import DAG, SubgraphDAGDelta, UpdateParametersDelta, apply_deltas_to_dag
+import numpy as np
+from aqueduct.artifacts import utils as artifact_utils
+from aqueduct.artifacts.artifact import Artifact
+from aqueduct.dag import DAG
+from aqueduct.enums import ArtifactType
 from aqueduct.error import AqueductError
-from aqueduct.generic_artifact import Artifact
 from aqueduct.utils import format_header_for_print, get_description_for_check
 
-from aqueduct import api_client
 
-
-class CheckArtifact(Artifact):
+class BoolArtifact(Artifact):
     """This class represents a check within the flow's DAG.
```

Comment:
Is this still representing a check, as this comment says? Or will any operator that returns a boolean be returned as a BoolArtifact. We should update this comment if its no-longer check exclusive.